pyllamacpp/src/main.cpp

```diff
@@ -447,7 +455,11 @@ int llama_generate(struct llama_context_wrapper * ctx_w, gpt_params params, py::
         if (!input_noecho) {
             for (auto id : embd) {
 //                printf("%s", llama_token_to_str(ctx, id));
-                new_text_callback(llama_token_to_str(ctx, id));
+                // If the host wants to stop generation, we should stop
+                if(!new_text_callback(llama_token_to_str(ctx, id))){
```

Comment:
You are calling `new_text_callback` two times. 
Is this the right behavior !?

Reply:
nope. 

Also this is not my code. Here is my code:
                new_text_callback(llama_token_to_str(ctx, id));
                if(!ctx_w->continue_gen){
                    llama_print_timings(ctx);
                    return 0;
                }

Reply:
Can you take a look at the Files changed tab of my pull request?
You seem to be looking at an old commit. I didn't rebase my commits before pushing so if you're not looking at the last one, you may find funny things I was testing.

And I didn't understand what do you mean by doing this in 1.0.7?